packages/playout-gateway/src/tsrHandlerJobs/initDeviceJob.ts

```diff
@@ -0,0 +1,201 @@
+import { Logger } from 'winston'
+import { Job } from './job'
+import {
+	AbortError,
+	CommandReport,
+	DeviceOptionsAny,
+	DeviceOptionsAtem,
+	DeviceStatus,
+	DeviceType,
+	MediaObject,
+	SlowFulfilledCommandInfo,
+	SlowSentCommandInfo,
+	StatusCode,
+} from 'timeline-state-resolver'
+import { TSRHandler } from '../tsrHandler'
+import { StatusObject } from '@sofie-automation/shared-lib/dist/peripheralDevice/peripheralDeviceAPI'
+import _ = require('underscore')
+import { CreateDeviceJobsResult } from './createDeviceJob'
+import { disableAtemUpload } from '../config'
+import { FinishedTrace, sendTrace } from '../influxdb'
+import debug = require('debug')
+
+export class InitDeviceJob extends Job<CreateDeviceJobsResult, CreateDeviceJobsResult, undefined> {
+	protected artifacts: undefined
+
+	constructor(
+		private deviceId: string,
+		private deviceOptions: DeviceOptionsAny,
+		private tsrHandler: TSRHandler,
+		private logger: Logger
+	) {
+		super()
+	}
+	async run(previousResult: CreateDeviceJobsResult, abortSignal?: AbortSignal): Promise<CreateDeviceJobsResult> {
+		if (abortSignal?.aborted) {
+			throw new AbortError()
+		}
+
+		const { deviceContainer, coreTsrHandler } = previousResult
+		// Set up device status
+		const deviceType = deviceContainer.deviceType
+
+		const onDeviceStatusChanged = (connectedOrStatus: Partial<DeviceStatus>) => {
+			let deviceStatus: Partial<StatusObject>
+			if (_.isBoolean(connectedOrStatus)) {
+				// for backwards compability, to be removed later
+				if (connectedOrStatus) {
```

Comment:
Could be a ternary

Reply:
+1 to Rasmus' Comment.

``` typescript
const deviceStatus = _.isBoolean(connectedOrStatus) 
  ? connectedOrStatus
    ? { statusCode: StatusCode.GOOD, }
    : {	statusCode: StatusCode.BAD, messages: ['Disconnected'], }
  : connectedOrStatus
```

Making it a double ternary, may or may not be wise.